File: app/src/main/java/com/paxus/pay/poslinkui/demo/utils/BundleMaker.java
package com.paxus.pay.poslinkui.demo.utils;

import android.os.Bundle;

public class BundleMaker {
    private Bundle bundle;

    public BundleMaker(){
        bundle = new Bundle();
    }

    public BundleMaker addLong(String key, long value){
        bundle.putLong(key, value);
        return this;
    }

    public BundleMaker addString(String key, String value){
        bundle.putString(key, value);
        return this;
    }

    public BundleMaker addBundle(Bundle bundle) {
        if(bundle != null) this.bundle.putAll(bundle);
        return this;
    }

    public Bundle get(){
        return bundle;
    }
}

File: app/src/main/java/com/paxus/pay/poslinkui/demo/utils/BundleMaker.kt
package com.paxus.pay.poslinkui.demo.utils

import android.os.Bundle

class BundleMaker {
    private val bundle: Bundle

    init {
        bundle = Bundle()
    }

    fun addLong(key: String?, value: Long): BundleMaker {
        bundle.putLong(key, value)
        return this
    }

    fun addString(key: String?, value: String?): BundleMaker {
        bundle.putString(key, value)
        return this
    }

    fun addBundle(bundle: Bundle?): BundleMaker {
        if (bundle != null) this.bundle.putAll(bundle)
        return this
    }

    fun get(): Bundle {
        return bundle
    }
}
